### PR TITLE
Teach k8s plugin how to kill a Pod

### PR DESCRIPTION
**Pull request by nemacysts**
This also adds in `namespace` as a required parameter for creating a k8s
executor instance.

### CODE REVIEW

**Comment by nemacysts** [comment]
we probably don't want to merge this until we have run() working so that we can actually run this code against a real cluster (at least until we have a working k8s itest setup in this repo)

File: task_processing/plugins/kubernetes/kubernetes_pod_executor.py [hunk 1]
```diff
@@ -67,7 +70,22 @@ def reconcile(self, task_config: KubernetesTaskConfig) -> None:
...
     def kill(self, task_id: str) -> bool:
-        pass
+        """
+        Delete a Pod by name.
+
+        This function will have task_processing stop tracking the killed task/Pod and
+        will return True if the Pod was successfully killed (and False otherwise)
+        """
+        with self._lock:
+            self.task_metadata = self.task_metadata.discard(task_id)
```
Comment (keymone) [inline task_processing/plugins/kubernetes/kubernetes_pod_executor.py:80]: we should probably only discard after we send request successfully and receive an event that confirms the deletion

Comment (nemacysts) [reply task_processing/plugins/kubernetes/kubernetes_pod_executor.py:80]: ah, I was trying to figure out the exact flow of how the mesos plugin handled this - that makes sense tho

Comment (nemacysts) [reply task_processing/plugins/kubernetes/kubernetes_pod_executor.py:80]: I guess I should also make sure that this delete call doesn't block too!

Comment (nemacysts) [reply task_processing/plugins/kubernetes/kubernetes_pod_executor.py:80]: I need to do some testing, but I think that the return value of a `delete_*` call is just if the k8s api successfully processed the request - not if the underlying object was deleted. 

so the only change that we'll want to make here is deferring the metadata deletion to when we have a Watch setup to get Pod events if I'm understanding thing correctly!

**Review by keymone — APPROVED** [review]
lgtm, i'd probably add a log line where we attempt to delete the pod for completeness, but not a blocker

**Pull request merged**